src/lua/api-gateway/validation/oauth2/oauthTokenValidator.lua

```diff
@@ -166,15 +166,8 @@ function _M:getTokenFromCache(cacheLookupKey)
     return nil;
 end
 
--- imsAuth will validate the service token passed in "Authorization" header --
-function _M:validate_ims_token()
+function _M:validateOAuthToken(oauth_token, validation_config)
```

Comment:
Yes, Lua is not OO friendly 😄 . If you'd rather define it as an instance member *I think* you can do something like:

```
OAuthTokenValidator:new({
  validation_config = foo
})
```
B/c the `new(o)` method is defined in the [Base Validator](https://github.com/adobe-apiplatform/api-gateway-request-validation/blob/master/src/lua/api-gateway/validation/validator.lua#L49-L54) and it decorates `o` with the methods and properties defined by that class.  

If you take this route please write a test to verify what I'm saying here, in case I'm wrong.